src/common/credentials.ts

```diff
@@ -120,7 +121,7 @@ export abstract class Credential extends Serializable{
 
   protected sigArray:Array<Signature> = [];
 
-  abstract getCredentialID():number;
+  abstract getCredentialID(codecID?:number):number;
```

Comment:
typescript question that I'm having a little trouble googling - does `codecID?:number` mean "maybe omitted, else a number"?

Reply:
Yes the `?` operator in arguments and object properties means it's optional. So in this case the `codecID` argument is optional but if it's implemented then it should be a `number`.